documentation/create_overview_doc/README.md

```diff
@@ -51,21 +51,27 @@ sh
 kt
 ```
 
+For flow map generation with Code2Flow, the tool currently supports the following languages, each with specific dependencies:
+
+- JavaScript: Requires Acorn for parsing. Ensure Acorn is installed and accessible in your environment.
+- Ruby: Uses Parser for parsing. Ensure the Parser gem is installed and accessible.
+- PHP: Employs PHP-Parser for parsing. PHP-Parser should be installed and accessible in your environment.
+- Python: No additional dependencies are required for Python projects.
+
+If your project uses a language not supported by Code2Flow for flow map generation, the documentation generation capabilities will still be available, but without the flow map visualization for those specific languages.
```

Comment:
**Optimization Issue**: The implementation details for language support in Code2Flow are mentioned, but there is no error handling for unsupported languages which could lead to failures or unhandled exceptions during the documentation generation process. <br> **Fix**: Implement error handling to manage cases where the project uses an unsupported language. This could involve skipping the flow map generation for those languages with a clear log message or providing a fallback mechanism. <br> **Code Suggestion**: 
 ```
 +if [ ! supported_languages.contains($language) ]; then
 +    echo "Warning: Language $language is not supported for flow map generation. Skipping..."
 +    continue
 +fi
 ```